src/main/java/com/relayrides/pushy/apns/ApnsConnection.java

```diff
@@ -254,7 +254,8 @@ protected void channelRead0(final ChannelHandlerContext context, final RejectedN
 			this.apnsConnection.sentNotificationBuffer.clearNotificationsBeforeSequenceNumber(rejectedNotification.getSequenceNumber());
 
 			final boolean isKnownBadRejection = this.apnsConnection.shutdownNotification != null &&
-					rejectedNotification.getSequenceNumber() == this.apnsConnection.shutdownNotification.getSequenceNumber();
+					(rejectedNotification.getSequenceNumber() == this.apnsConnection.shutdownNotification.getSequenceNumber()
+					|| rejectedNotification.getSequenceNumber() == 0);
```

Comment:
It's probably unnecessary caution, but it might be good to check the rejection reason here, too:

```java
// ...
|| (rejectedNotification.getSequenceNumber() == 0 && RejectedNotificationReason.MISSING_TOKEN.equals(rejectedNotification.getReason()))
```